tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -188,8 +188,10 @@ constify_tableoid_walker(Node *node, ConstifyTableOidContext *ctx)
 		 * we doublecheck system columns here because projection will
 		 * segfault if any system columns get through
 		 */
-		if (var->varattno < 0)
-			elog(ERROR, "transparent decompression only supports tableoid system column");
+		if (var->varattno < SelfItemPointerAttributeNumber)
+			elog(ERROR,
+				 "transparent decompression only supports tableoid system column %d",
```

Comment:
this message should probably be reverted or made more explcit that the %d attno is not supported